modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+FOR (actor:Actor) REQUIRE actor.surname IS NODE KEY
+OPTIONS {
+  indexProvider: 'range-1.0'
+}
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+.Create a relationship property uniqueness constraint with a specified index provider
+======
+
+.Create a constraint requiring `SEQUEL_OF` relationships to have a unique combination of `order`, `seriesTitle`, and `number` properties, specifying `range-1.0` as index provider
+[source, cypher]
+----
+CREATE CONSTRAINT rel_constraint_with_options
+FOR ()-[sequel:SEQUEL_OF]-() REQUIRE (sequel.order, sequel.seriesTitle, sequel.number) IS UNIQUE
+OPTIONS {
+  indexProvider: 'range-1.0'
+}
+----
+
+.Result
+[source, queryresult]
+----
+Added 1 constraint.
+----
+
+======
+
+There are no valid index configuration values for the constraint-backing range indexes.
+
+[[create-a-constraint-if-not-exist]]
+=== Handling existing constraints when creating a constraint
+
+Creating an already existing constraint will fail.
+To avoid such an error, `IF NOT EXISTS` can be added to the `CREATE` command.
+This will ensure that no error is thrown and that no constraint is created if any other constraint with the given name, or another node property uniqueness constraint on the same schema, already exists.
+As of Neo4j 5.17, an informational notification is instead returned showing the existing constraint which blocks the creation.
+
+.Create a relationship property uniqueness constraint when a constraint with the same name exists
+======
+
+.Create a constraint requiring all `SEQUEL_OF` relationships to have unique `order` properties
+[source, cypher]
+----
+CREATE CONSTRAINT sequels IF NOT EXISTS
+FOR ()-[sequel:SEQUEL_OF]-() REQUIRE sequel.order IS UNIQUE
+----
+
+Because the same constraint already exists, nothing will happen:
+
+.Result
+[source, queryresult]
+----
+(no changes, no records)
+----
+
+.Notification
+[source]
+----
+`CREATE CONSTRAINT sequels IF NOT EXISTS FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order) IS UNIQUE` has no effect.
+`CONSTRAINT sequels FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order, e.seriesTitle) IS UNIQUE` already exists.
+----
+
+======
+
+
+[[constraint-creation-failures-and-data-violations]]
+=== Constraint creation failures and data violation scenarios
+
+* xref:constraints/managing-constraints.adoc#create-an-already-existing-constraint[]
+* xref:constraints/managing-constraints.adoc#create-data-that-violates-a-constraint[]
+* xref:constraints/managing-constraints.adoc#fail-to-create-constraint-due-to-existing-data[]
+* xref:constraints/managing-constraints.adoc#removing-an-existing-constrained-property-will-fail[]
+* xref:constraints/managing-constraints.adoc#create-key-and-uniqueness-constraint-on-same-schema-as-existing-index[]
+
+
+[[create-an-already-existing-constraint]]
+==== Creating an already existing constraint will fail
+
+Creating a constraint on a node label or relationship property that is already constrained will fail.
```

Comment:
this sentence feels like it's missing some words?
```suggestion
Creating a constraint on the same label or relationship type and properties that is already constrained will fail.
```

Reply:
might also be nice to add something about the property type being the same for the property type constraint case